add mysql_optional as dsn parameter

Thomblin opened the pull request:
This pull requests adds `mysql_ssl_option` as an dsn paramter to pt-online-schema-change.

Why?

Upgrading debian from bullseye to bookworm or trixie changes the behaviour of perl DBI. As a consequence perl DBI is not able to use ssl anymore when enforced (pt-online-schema-change raises the exception "SSL connection error: Enforcing SSL encryption is not supported"). 

See: [Debian Bug Report 1032074](https://bugs-devel.debian.org/cgi-bin/bugreport.cgi?bug=1032074)

I found no fix for this (using official sources). The only workaround to make it work is calling pt-online-schema-change with both `mysql_ssl=1` and `mysl_ssl_optional=1`

Please advice how and if tests or documentation need to be adjusted.

- [X] The contributed code is licensed under GPL v2.0
- [X] Individual Contributor License Agreement is signed
- [ ] util/update-modules has been ran
- [ ] Documentation updated
- [ ] Test suite update

Thomblin committed:
add mysql_optional as dsn parameter

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/signed)](https://cla-assistant.percona.com:443/percona/percona-toolkit?pullRequest=1022) <br/>All committers have signed the CLA.

svetasmirnova commented:
Immediate workaround is to install DBD::mysql, depending on upstream or Percona libmysql, available in Percona repositories.

I will check proposed solution, so Percona Toolkit can run with default DBD::mysql, linked to libmariadb3 too. Currently I can only say that you modified auto-generated code that will be rewritten once we create packages or simply update modules. Check CONTRIBUTING.md for how to update modules and create tests for your contributions.

Thomblin committed:
add parameter mysql_optional

Thomblin commented:
@svetasmirnova I added the option to lib and tests

svetasmirnova committed:
PR-1022 - add mysql_optional as dsn parameter#1022

- Adjusted t/pt-online-schema-change/ssl.t, so it checks if secure connection works
- Checked --help output for --mysql_ssl_optional together with other options in option_sanity.t

svetasmirnova committed:
PR-1022 - add mysql_optional as dsn parameter#1022

- Added ssl_optional to defaul Sandbox connection string
- Added new test case t/pt-online-schema-change/ssl_optional.t for
  MariaDB-based DBD::mysql
- Added check in ssl.t, so it runs for MySQL-based DBD::mysql only

svetasmirnova committed:
PR-1022 - add mysql_optional as dsn parameter#1022

- Updated modules in all tools

svetasmirnova committed:
PR-1022 - add mysql_optional as dsn parameter#1022

- Updated help and magic options in other tools affected
- Updated tests in some of other tools

svetasmirnova committed:
PR-1022 - add mysql_optional as dsn parameter#1022

- Added options when they were missed
- Updated ssl.t tests

svetasmirnova committed:
PR-1022 - add mysql_optional as dsn parameter#1022

- MariaDB checks better location in ssl.t tests

svetasmirnova committed:
PR-1022 - add mysql_optional as dsn parameter#1022

- Added MariaDB library tests
